### PR TITLE
Jessica Nascimento

### PR DESCRIPTION
**Pull request by Jessica7**

### CODE REVIEW

**Comment by brunoventura** [comment]
Please, add vendors (bower_components) folder to gitignore

File: project/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "gulp": "^3.9.1",
+    "gulp-concat": "^2.6.0",
+    "gulp-html5-lint": "^1.0.1",
+    "gulp-htmlmin": "^1.3.0",
+    "gulp-rename": "^1.2.2",
+    "gulp-sass": "^2.1.0",
+    "gulp-scss-lint": "^0.3.9",
+    "gulp-sourcemaps": "^1.6.0",
+    "gulp-uglify": "^1.5.3",
+    "n": "^2.0.2",
```
Comment (brunoventura) [inline project/package.json]: You need that node version manager package in the project?

File: project/dev/frontend/js/helpers/utils.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+   * @param {async} boolean
+   * @param {Error} function
+   */
+	var _this = this;
+	this.method =  conf.method || 'GET';
+	this.url = conf.url || '';
+	this.data = conf.data || null;
+	this.success = conf.success || function(){};
+	this.async = conf.async || true;
+  this.Error = conf.Error || function(){};
```
Comment (brunoventura) [inline project/dev/frontend/js/helpers/utils.js]: Why only this conf param is capitalized?

File: project/dev/frontend/js/main.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  /**
+   * listUrls() - A method that receives request by parameter data and make
+   * list of links and return all links.
+   * @param {data} arr
+   */
+  function listUrls(data) {
+    var $topLinks = MY_UTILS.$dom.el('.link-list__all-items');
+    var fragment;
+    fragment = document.createDocumentFragment();
+    JSON.parse(data).map(function(item, limitShowLinks) {
```
Comment (brunoventura) [inline project/dev/frontend/js/main.js]: Are you not parsing the json in the $ajax.dataHttp method?

File: project/dev/frontend/js/main.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+      if (limitShowLinks < 5) fragment.appendChild(makeList(item));
```
Comment (brunoventura) [inline project/dev/frontend/js/main.js]: I think that's not a good practice call the index from map as "limitShowLinks".
You can do this for more readable code:
````javascript
var limitShowLinks = 5;
data.map(function(item, index) {
  if (index < limitShowLinks) fragment.appendChild(makeList(item));
});
````

File: project/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	  sourcemaps = require('gulp-sourcemaps'),
+		uglify = require('gulp-uglify'),
+		rename = require('gulp-rename'),
+		concat = require('gulp-concat'),
+		htmlmin = require('gulp-htmlmin');
+
+gulp.task('build', function(){
+	gulp.src(['./dev/frontend/images/**/*.{jpg,png,gif}'])
+	.pipe(gulp.dest('./dist/images'));
+	gulp.src('./dev/frontend/vendors/**')
```
Comment (brunoventura) [inline project/gulpfile.js]: I think you can set bowerrc directory directly to dist folder, I don't see any usage to justify the copy of files in every build task

File: project/dev/frontend/js/main.js (hunk 3 above)
Comment (brunoventura) [inline project/dev/frontend/js/main.js]: are you sorting the links?

**Comment by Jessica7** [comment]
Thanks for your feedback  @brunoventura , I loved participating in the challenge was an excellent experience and learning :+1:

**Comment by brunoventura** [comment]
Evaluating challenges we always learned a lot too :) If one day come to Floripa, visit us.

Thank you.

**Comment by brunoventura** [comment]
Etherpad?

On Tue, Apr 5, 2016 at 10:37 PM, Jéssica Nascimento <
notifications@github.com> wrote:

> Thanks for your feedback @brunoventura <https://github.com/brunoventura>
> , I loved participating in the challenge was an excellent experience and
> learning [image: :+1:]
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/chaordic/frontend-intern-challenge/pull/8#issuecomment-206067204>
>



-- 
[image: Linx | Neemu + Chaordic]

BRUNO VENTURA

engineering

 +55 48 3232.3200  +55 48 9157.5933
 Av. Afonso Delambert Neto, 26 - Lagoa da Conceição
Florianópolis, SC - Brasil - 88062-000

**Pull request closed without merging**